server/coordinator/factory.go

```diff
@@ -187,10 +187,14 @@ func (f *Factory) makeCreatePartitionTableProcedure(ctx context.Context, request
 	})
 }
 
-func (f *Factory) CreateDropTableProcedure(ctx context.Context, request DropTableRequest) (procedure.Procedure, error) {
+// CreateDropTableProcedure creates a procedure to do drop table.
+//
+// And if no error is thrown, the returned boolean value is used to tell whether the procedure is created.
+// In some cases, e.g. the table doesn't exist, it should not be an error and false will be returned.
+func (f *Factory) CreateDropTableProcedure(ctx context.Context, request DropTableRequest) (procedure.Procedure, bool, error) {
 	id, err := f.allocProcedureID(ctx)
 	if err != nil {
-		return nil, err
+		return nil, false, err
```

Comment:
```suggestion
		return nil, true, err
```
Returning false means that this error will not be passed to the user. If alloc table id fails, we need to pass this error out.